AK/FlyString.cpp

```diff
@@ -15,20 +15,10 @@ namespace AK {
 
 static auto& all_fly_strings()
 {
-    static Singleton<HashMap<StringView, uintptr_t>> table;
+    static Singleton<HashMap<StringView, Detail::StringBase>> table;
```

Comment:
Is storing these as values rather than opaque pointers the reason for the awkward refcount dance below in this commit? 

... thinking about it a bit longer, if the intention is to keep FlyStrings' data around forever is there any point in messing with refcounts at all in FlyString instance creation/destruction? I would expect we can get away with checking if all fly strings has our data, and if it does, just share that StringBase with the table as the 'true owner'

Reply:
> if the intention is to keep FlyStrings' data around forever

No, what this code is trying to do (and failing to do cleanly) is to store `WeakPtr<StringData>`.  The best way to do so (in terms of readability) is, obviously, to implement it and use here. While it might be nicer than Andreas' original approach that used a ton of manual refcounting, I doubt it would be better than my five lines of "refcount dance". It obviously would take more than five lines and, crucially, I don't see how a non-owning pointer to string data can be useful anywhere else.

Reply:
Gotcha. All right, we're good here then, this is essentially the same as the old impl from what you just said.